craft_application/models/manifest.py

```diff
@@ -14,19 +14,35 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """Models representing manifests for projects and fetch-service assets."""
+import hashlib
+import pathlib
+from datetime import datetime, timezone
 from typing import Any, Literal
 
 from pydantic import Field
 from typing_extensions import override
```

Comment:
```suggestion
from typing_extensions import Self, override
```

(This is just so if you want you can web-merge my suggestion below)